### DISCUSSION TITLE
Recipe for migrating from JUL to slf4j? With support for `Logger.log(Level,String)` calls?

### DISCUSSION THREAD

**Discussion by woj-tek** [openrewrite/rewrite#4249] [Q&A] [answered]
Hi,
For a long time we were using JUL (`java.util.Logger`, I know, bad!) and we kinda got stuck with it. However, it's getting more and more annoying to deal with it and it's high time to migrate to slf4j. I was looking for recipes to do so but found none.

I found, somewhat related, [Migrate JUL to Log4j 2.x API](https://docs.openrewrite.org/recipes/java/logging/log4j/jultolog4j) (with the hope to use [Migrate Log4j to SLF4J](https://docs.openrewrite.org/recipes/java/logging/slf4j/log4jtoslf4j)) but the first one didn't work because we rely heavily on `Logger.log(Level,[message])` API instead of `Logger.<level>(<message>)` one which seems to be supported by looking at Definition and method signature.

Does anyone know any recipe for `Logger.log(Level,String)` calls?

**Comment by timtebeek** [comment, marked as answer]
hi! Sounds like something that we can fairly easily cover with recipes; let's discuss this further in the dedicated logging module:
- https://github.com/openrewrite/rewrite-logging-frameworks/issues/155
